### PR TITLE
Feature/long markup label

### PR DESCRIPTION
**Pull request by brosenberg42**
**Issues:**
- https://github.com/openmpf/openmpf/issues/1526

- - -
This change is [Reviewable](https://reviewable.io/reviews/openmpf/openmpf/1579)

### CODE REVIEW

**Review by jrobble — CHANGES_REQUESTED** [review]
> Reviewed 22 of 22 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/openmpf/openmpf/1579)* status: all files reviewed, 6 unresolved discussions (waiting on @brosenberg42 and @clnowacki)

*[a discussion](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjkFdU0UnSEaDvxOiF:-NAjkFdU0UnSEaDvxOiG:b-9decvf) (no related file):*
Please update the Markup Guide.

___
___
*[`trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java` line 208 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjnUHBDWMPPMrZqaca:-NAjnUHCA-eqzR8QkvdY:bbptvn3) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java#L208)):*
> ```Java
>         int labelMaxLength;
>         try {
>             labelMaxLength = Integer.parseInt(markupProperties.get(
> ```

Later on we do `textPart.substring(0, textLength).strip();`. According to the docs, this will throw a `StringIndexOutOfBoundsException` if `textLength` is negative.
___
*[`trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java` line 214 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjjEBWEsMyufkb8a4U:-NAjjEBWEsMyufkb8a4V:b9anbuo) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java#L214)):*
> ```Java
>             labelMaxLength = 10;
>             var errorMsg = String.format(
>                 "Could not convert the value of the %s property to an integer. Using %s instead.",
> ```

Should we log the invalid value for the user's awareness?
___
*[`trunk/workflow-manager/src/main/resources/properties/mpf.properties` line 378 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjjwCS-J4HHTv_uWsn:-NAjjwCS-J4HHTv_uWso:b-i36lwl) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/resources/properties/mpf.properties#L378)):*
> ```INI
> # Leave empty to omit.
> #markup.labels.text.prop.to.show=CLASSIFICATION
> markup.labels.text.prop.to.show=TEST
> ```

I think you changed this for debug purposes and it should be changed back.
___
*[`trunk/workflow-manager/src/main/resources/properties/mpf.properties` line 382 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjlIKqDszNirgfFQ61:-NAjlIKqDszNirgfFQ62:b-s7eiii) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/resources/properties/mpf.properties#L382)):*
> ```INI
> # The maximum length of the label selected by markup.labels.text.prop.to.show. If the label is
> # longer than the limit, characters after limit will not be displayed.
> markup.labels.text.max.length=10
> ```

I noticed that the https://github.com/openmpf/openmpf/blob/feature/long-markup-label/trunk/markup/src/main/resources/markup.properties file is empty. Should we remove it and the reference to it [here](https://github.com/openmpf/openmpf/blob/feature/long-markup-label/trunk/markup/src/main/resources/appConfig.xml#L35)?
___
*[`trunk/workflow-manager/src/main/resources/templates/Algorithms.json` line 45 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjlAIL9bOo37aZd78r:-NAjlAIL9bOo37aZd78s:b4ddfd0) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/resources/templates/Algorithms.json#L45)):*
> ```JSON
>                 },
>                 {
>                     "name": "MARKUP_TEXT_LABEL_MAX_LENGTH",
> ```

Please also update the `Algorithms.json` file that's part of the system tests and custom system tests.

File: trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java [hunk 1]
```diff
@@ -206,6 +203,21 @@ private List<Markup.BoundingBoxMapEntry> createMapEntries(
         var animate = Boolean.parseBoolean(markupProperties.get(
```
Comment (brosenberg42) [inline trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java:203]: test

**Review by brosenberg42 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/openmpf/openmpf/1579)* status: 18 of 25 files reviewed, 6 unresolved discussions (waiting on @clnowacki and @jrobble)

*[a discussion](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjkFdU0UnSEaDvxOiF:-NAoH1nIEJPpVSWbYFHa:b-896fix) (no related file):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

Please update the Markup Guide.
</blockquote></details>

Done.

___
___
*[`trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java` line 208 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjnUHBDWMPPMrZqaca:-NAo2gFX3Vjp815E6X2q:b-896fix) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java#L208)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

Later on we do `textPart.substring(0, textLength).strip();`. According to the docs, this will throw a `StringIndexOutOfBoundsException` if `textLength` is negative.
</blockquote></details>

Done.
___
*[`trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java` line 214 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjjEBWEsMyufkb8a4U:-NAo2gedExBvYNyoJiC5:b-896fix) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/markup/MarkupSplitter.java#L214)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

Should we log the invalid value for the user's awareness?
</blockquote></details>

Done.
___
*[`trunk/workflow-manager/src/main/resources/properties/mpf.properties` line 378 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjjwCS-J4HHTv_uWsn:-NAo3QBRDI_TvJe-kz2z:b-896fix) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/resources/properties/mpf.properties#L378)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

I think you changed this for debug purposes and it should be changed back.
</blockquote></details>

Done.
___
*[`trunk/workflow-manager/src/main/resources/properties/mpf.properties` line 382 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjlIKqDszNirgfFQ61:-NAo3QXADnaf0vuLfFkn:b-896fix) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/resources/properties/mpf.properties#L382)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

I noticed that the https://github.com/openmpf/openmpf/blob/feature/long-markup-label/trunk/markup/src/main/resources/markup.properties file is empty. Should we remove it and the reference to it [here](https://github.com/openmpf/openmpf/blob/feature/long-markup-label/trunk/markup/src/main/resources/appConfig.xml#L35)?
</blockquote></details>

Done.
___
*[`trunk/workflow-manager/src/main/resources/templates/Algorithms.json` line 45 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjlAIL9bOo37aZd78r:-NAo4gMi29CQWP8qVhUY:b-896fix) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/workflow-manager/src/main/resources/templates/Algorithms.json#L45)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

Please also update the `Algorithms.json` file that's part of the system tests and custom system tests.
</blockquote></details>

Done.

**Review by jrobble — CHANGES_REQUESTED** [review]
> Reviewed 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/openmpf/openmpf/1579)* status: all files reviewed, 1 unresolved discussion (waiting on @brosenberg42 and @clnowacki)

___
*[`trunk/video-overlay/src/main/cpp/BoundingBoxWriter.cpp` line 440 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjiJ0v0wD670XjeixS:-NAjiJ0v0wD670XjeixT:beatom) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/video-overlay/src/main/cpp/BoundingBoxWriter.cpp#L440)):*
> ```cpp
>     // Calculate frame padding for worst-case scenario.
>     int baseline = 0;
>     std::string maxSizeLabel(std::max(maxLabelLength, 1), 'W'); // "W" is the widest character
> ```

This no longer takes into consideration the widest numeric part since you removed `888.888`. I would assume this can result in some labels extending off the screen when the border is enabled.

**Review by brosenberg42 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/openmpf/openmpf/1579)* status: all files reviewed, 1 unresolved discussion (waiting on @clnowacki and @jrobble)

___
*[`trunk/video-overlay/src/main/cpp/BoundingBoxWriter.cpp` line 440 at r1](https://reviewable.io/reviews/openmpf/openmpf/1579#-NAjiJ0v0wD670XjeixS:-NAoNvZH5cD4fZ0aVLdI:brxsz95) ([raw file](https://github.com/openmpf/openmpf/blob/57986f403130a336138987d81cb4057d41e646da/trunk/video-overlay/src/main/cpp/BoundingBoxWriter.cpp#L440)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

This no longer takes into consideration the widest numeric part since you removed `888.888`. I would assume this can result in some labels extending off the screen when the border is enabled.
</blockquote></details>

That value includes the numbers. It just uses W for the numbers too.

**Review by jrobble — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/openmpf/openmpf/1579)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @clnowacki)

**Review by jrobble — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/openmpf/openmpf/1579)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @clnowacki)

**Review by clnowacki — APPROVED** [review]
> Reviewed 4 of 22 files at r1, 4 of 7 files at r2, 1 of 1 files at r3, 10 of 15 files at r4, 6 of 6 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/openmpf/openmpf/1579)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @brosenberg42)

**Pull request merged**